src/CMakeLists.txt

```diff
@@ -143,6 +143,17 @@ if(APPLE AND (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR CMAKE_CXX_COMPILER_
         list(REMOVE_DUPLICATES CMAKE_CXX_IMPLICIT_LINK_LIBRARIES)
 endif()
 
+# Check for base threading library
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+find_package(Threads REQUIRED)
+if(${CMAKE_THREAD_LIBS_INIT} MATCHES "-pthread")
```

Comment:
Looking at https://github.com/Kitware/CMake/blob/v3.1.3/Modules/FindThreads.cmake, i guess `THREADS_PREFER_PTHREAD_FLAG` + `Threads::Threads` import target would simplify it.
But we depend on `cmake-3.0`, and that would require bumping dep to 3.1;
Can not say whether it would be ok or not